homeassistant/components/freedompro/fan.py

```diff
@@ -87,27 +88,30 @@ async def async_added_to_hass(self) -> None:
         await super().async_added_to_hass()
         self._handle_coordinator_update()
 
-    async def async_turn_on(self, percentage=None, preset_mode=None, **kwargs):
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
         """Async function to turn on the fan."""
         payload = {"on": True}
-        payload = json.dumps(payload)
         await put_state(
             self._session,
             self._api_key,
             self.unique_id,
-            payload,
+            json.dumps(payload),
```

Comment:
mypy doesn't like reassigning a new type (`str`) to the `dict` variable, so I've moved the `json.dumps` routine accordingly.